pkg/languages/js/js.go

```diff
@@ -153,13 +194,20 @@ func generatePackageJson(configurationEntries map[string][]string) ([]byte, erro
 		for _, dep := range deps {
 			splitted := strings.Split(dep, " ")
 			if len(splitted) != 2 {
-				return nil, fmt.Errorf("Invalid dependency entry: %v", dep)
+				return nil, fmt.Errorf("invalid dependency entry: %v", dep)
 			}
 			depsRoot[strings.Trim(splitted[0], " ")] = strings.Trim(splitted[1], " ")
 		}
 	}
 
+	root["scripts"] = map[string]string{
+		"test": "tape tests/**/*.js",
+	}
+
 	root["dependencies"] = depsRoot
+	root["devDependencies"] = map[string]string{
+		"tape": "4.11.0",
+	}
```

Comment:
I'm not sure about this. I personally like `tape` but I don't know if we want to endorse it in this way. Seems to me that if the user wants tests, requiring that to be defined in a `package.json` is going to be best.

Reply:
I think that we should provide one by default that is most common in js world, so you can do zero-config tests without requiring any package.json
Then, if you want a custom one, or you define an env somewhere (like comments) that we support (like test-framework or something like this), or you write your own package.json

Reply:
The problem is that the JavaScript ecosystem is pretty fractured and there isn't a single standout test framework that everyone uses. Here for example are the top 10 in 2019 with a bunch of honorable mentions as well (including `tape`). So choosing which framework to expose is going to be difficult.

Node.js itself exposes an `assert` module, so it's possible to test without using a framework at all. https://nodejs.org/api/assert.html

Maybe a good approach is to just run `node index_test.js` as the test task in the generated `package.json`. This allows for basic testing using the built in `assert` module, but avoids making a choice on one of the many frameworks out there.

Reply:
> Maybe a good approach is to just run node index_test.js as the test task in the generated package.json

I agree with the idea to "don't endorse a specific test framework", but assert module itself provides only assertions and not test runtime right?

I think we can add a configuration flag like `test-framework`:

```
// kfn:test-framework tape
```

And just support the famous ones (jest, tape, mocha). And by default, just choose one of those. If you want to manually configure it, we could support something like:

```
// kfn:dev-dependency tape latest
// kfn:npm-script test tape tests/**/*.js 
```

Or the user creates by hand the `package.json`

Reply:
Let me try to implement it so i explain myself better